use standard claims (#16)
thomas-fossati/draft-psa-token: draft-tschofenig-rats-psa-token.md

protocols, including secure provisioning and network access control.  This
document specifies the PSA attestation token structure and semantics.

The PSA attestation token is a profiled Entity Attestation Token (EAT).

This specification describes what claims are used in an attestation token
generated by PSA compliant systems, how these claims get serialized to the

> wire, and how they are cryptographically protected.

Reviewer: Does this have a specific meaning? We don't make any statements about endianess etc

Author: We "inherit" the serialisation rules of CBOR.

## Caller Claims

### Nonce
{: #sec-nonce-claim}

> The Nonce claim is a challenge from the caller.

Reviewer 2: ```suggestion
The Nonce claim is used to carry the challenge provided by the caller to demonstrate freshness of the generated token.
```

The EAT {{!I-D.ietf-rats-eat}} `nonce` (claim key 10) is used.  The following
constraints apply to the `nonce-type`:

> * The length MUST be 32, 48, or 64 bytes;

Reviewer 2: ```suggestion
* The length MUST be either 32, 48, or 64 bytes.
```

> * There MUST be only one nonce value.

Reviewer 2: ```suggestion
* Only a single nonce value is conveyed. Per {{!I-D.ietf-rats-eat}} the array notation is not used for encoding the nonce value.
```

> Note that the latter implies that the array form MUST NOT be used.

Reviewer 2: ```suggestion

```

Reviewer 2: I moved the text

## Target Identification Claims

### Instance ID
{: #sec-instance-id-claim}

The Instance ID claim represents the unique identifier of the Initial
Attestation Key (IAK).  The full definition is in {{PSA-SM}}.

The EAT `ueid` (claim key 11) of type RAND is used.  The following constraints
apply to the `ueid-type`:

> * The length MUST be 33 bytes;

Reviewer 2: ```suggestion
* The length MUST be 33 bytes.
```

## Target Identification Claims

### Instance ID
{: #sec-instance-id-claim}

The Instance ID claim represents the unique identifier of the device.

> It is a 32 bytes hash of the public key corresponding to the Initial

Reviewer 3: The current PSA Security model has backed off from this strict definition of the Instance ID. It now just says it 'identifies the IAK'.

Reviewer: We'll also want to revise it to also cover group unique keys as an alternative

Reviewer: Unless there is a separate, distinct EAT claim for that?

Author: @SimonFrost-Arm : makes sense, I will remove the details about the derivation.

@adrianlshaw : can you suggest some text?

Author: > Unless there is a separate, distinct EAT claim for that?

There is some discussion ongoing about defining a separate "group ueid" coming from the FIDO side.

Reviewer: On a re-read, I think 'identifies the IAK' should be fine for both cases.

Author: done in 58ba0c6

~~~

### Profile Definition
{: #sec-profile-definition-claim}

The Profile Definition claim encodes the unique identifier that corresponds to
the EAT profile described by this document.  This allows a receiver to assign
the intended semantics to the rest of the claims found in the token.

The EAT `profile` (claim key 18) is used.  The following constraints
apply to its type:

> * The URI encoding MUST be used;

Reviewer 2: ```suggestion
* The URI encoding MUST be used.
```

* The value MUST be `http://arm.com/psa/2.0.0`.

This claim MUST be present in a PSA attestation token.

See {{sec-backwards-compat}}, for considerations about backwards compatibility
with previous versions of the PSA attestation token format.

~~~
{::include cddl/psa-profile.cddl}
~~~

# Backwards Compatibility Considerations
{: #sec-backwards-compat}

> Previous versions of this specification used different keys for the following

Reviewer 2: ```suggestion
Previous versions of this specification used different claim key values for the following
```

claims:

* Nonce (claim key -75008);
* Instance ID (claim key -75009);
* Profile Description (claim key -75000 and value `PSA_IOT_PROFILE_1`).

These claim keys have been retired and their use is deprecated.

Unless compatibility with existing infrastructure is a concern, emitters (e.g.,
devices that implement the PSA Attestation API) SHOULD produce tokens with
their standard equivalent instead, as described in {{sec-nonce-claim}},
{{sec-instance-id-claim}} and {{sec-profile-definition-claim}} respectively.

> In order to smooth the transition to the token format described in this

Reviewer 2: ```suggestion
To simplify the transition to the token format described in this
```

document it is RECOMMENDED that receivers (e.g., PSA Attestation Verifiers)
accept tokens encoded according to the old profile (`PROFILE_IOT_1`) as well as
to the new profile (`http://arm.com/psa/2.0.0`), at least for the time needed to
their clients to upgrade.

# Token Encoding and Signing

The PSA attestation token is encoded in CBOR {{!RFC7049}} format.  Only
definite-length string, arrays, and maps are allowed.

Cryptographic protection is obtained by wrapping the `psa-token` map in a COSE
Web Token (CWT) {{!RFC8392}}.  For asymmetric key algorithms, the signature
structure MUST be COSE_Sign1.  For symmetric key algorithms, the signature
structure MUST be COSE_Mac0.

The CWT CBOR tag (61) is not used.  An application that needs to exchange PSA

> attestation tokens can use the Media Type defined in {{sec-iana-media-types}}

Reviewer 2: ```suggestion
attestation tokens can use the media type defined in {{sec-iana-media-types}}
```

~~~

### Profile Definition
{: #sec-profile-definition-claim}

The Profile Definition claim encodes the unique identifier that corresponds to
the EAT profile described by this document.  This allows a receiver to assign
the intended semantics to the rest of the claims found in the token.

The EAT `profile` (claim key 18) is used.  The following constraints
apply to its type:

* The URI encoding MUST be used;
* The value MUST be `http://arm.com/psa/2`.

This claim MUST be present in a PSA attestation token.

See {{sec-backwards-compat}}, for considerations about backwards compatibility
with previous versions of the PSA attestation token format.

~~~
{::include cddl/psa-profile.cddl}
~~~

# Backwards Compatibility Considerations
{: #sec-backwards-compat}

Previous versions of this specification used different keys for the following
claims:

* Nonce (claim key -75008);
* Instance ID (claim key -75009);
* Profile Description (claim key -75000 and value `PSA_IOT_PROFILE_1`).

> These claim keys have been retired and their use is deprecated.

Reviewer 3: Maybe worth saying this choice was taken because IANA registration became available for standard claim values (rather than being an arbitrary decision we took for fun)!

Author: I had a sentence to that end, but I dropped it because I thought that it'd be a distraction.  We want cold, dull, uninspiring standards language :-)

Reviewer: > I had a sentence to that end, but I dropped it because I thought that it'd be a distraction. We want cold, dull, uninspiring standards language :-)
This

Unless compatibility with existing infrastructure is a concern, emitters (e.g.,

> devices that implement the PSA attestation API) SHOULD produce tokens with

Reviewer: missing capital A in PSA Attestation API

Author: done in 58ba0c6

their standard equivalent instead, as described in {{sec-nonce-claim}},
{{sec-instance-id-claim}} and {{sec-profile-definition-claim}} respectively.

In order to smooth the transition to the token format described in this
document it is RECOMMENDED that receivers (e.g., PSA Attestation Verifiers)
accept tokens encoded according to the old profile (`PROFILE_IOT_1`) as well as
to the new profile (`http://arm.com/psa/2`), at least for the time needed to
their clients to upgrade.

# Token Encoding and Signing

The PSA attestation token is encoded in CBOR {{!RFC7049}} format.  Only
definite-length string, arrays, and maps are allowed.

Cryptographic protection is obtained by wrapping the `psa-token` map in a COSE
Web Token (CWT) {{!RFC8392}}.  For asymmetric key algorithms, the signature
structure MUST be COSE_Sign1.  For symmetric key algorithms, the signature
structure MUST be COSE_Mac0.

The CWT CBOR tag (61) is not used.  An application that needs to exchange PSA
attestation tokens can use the Media Type defined in {{sec-iana-media-types}}
or the CoAP Content-Format defined in {{sec-iana-coap-content-format}}.

# Freshness Model

> The PSA Token supports the freshness models for attestation Evidence based on

Reviewer: Should evidence be capitalised like this?

Author: It's the term of art established in the [RATS architecture document](https://ietf-rats-wg.github.io/architecture/draft-ietf-rats-architecture.html#section-8.1)
